app/assets/javascripts/course/assessment/submission/submission.js

```diff
@@ -164,14 +164,21 @@
       $('#' + identifier).find('textarea.code').ace();
     });
 
-    // Show the first tab on page load.
-    $(MULTI_QUESTION_ASSESSMENT_SELECTOR + '.tab-header a:first').tab('show');
+    var step = getParamFromURL('step=') || 1;
+    $(MULTI_QUESTION_ASSESSMENT_SELECTOR + '.tab-header a[step="' + step + '"]').tab('show');
   }
 
   function initializeAceEditor() {
     $(DOCUMENT_SELECTOR + 'textarea.code').not('.tab-content textarea.code').ace();
   }
 
+  function getParamFromURL(param) {
```

Comment:
@fonglh current implementation won't work if there are multiple params, because you are splitting by `step=`, e.g. if the params is `step=1&xxx=2`.